fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/facades/SimulationFacade.java

```diff
@@ -492,35 +496,42 @@ private SumoVehicleState processVehicleSubscriptionResult(final long time,
             return null;
         }
 
-        final VehicleData.Builder vehicleDataBuilder = new VehicleData.Builder(time, veh.id)
-                .position(veh.position.getGeographicPosition(), veh.position.getProjectedPosition())
-                .movement(veh.speed, veh.acceleration, fixDistanceDriven(veh.distanceDriven, sumoVehicle.lastVehicleData))
-                .orientation(DriveDirection.UNAVAILABLE, veh.heading, veh.slope)
-                .route(veh.routeId)
-                .signals(decodeVehicleSignals(veh.signalsEncoded))
-                .stopped(vehicleStopMode)
-                .sensors(createSensorData(sumoVehicle, veh.leadingVehicle, veh.followerVehicle, veh.minGap))
-                .laneArea(vehicleSegmentInfo.get(veh.id));
+        final VehicleData.Builder vehicleDataBuilder = new VehicleData.Builder(time, veh.id);
 
-
-        if (isParking) {
-            if (!sumoVehicle.lastVehicleData.isStopped()) {
-                log.info("Vehicle {} has parked at {} (edge: {})", veh.id, veh.position, veh.edgeId);
-            }
-            vehicleDataBuilder
-                    // use the last known road position, otherwise we can not retrieve a valid one
-                    .road(sumoVehicle.lastVehicleData.getRoadPosition())
-                    // for parking vehicles, there are no consumptions and emissions to measure
-                    .consumptions(new VehicleConsumptions(
-                            new Consumptions(0d), sumoVehicle.lastVehicleData.getVehicleConsumptions().getAllConsumptions())
-                    ).emissions(new VehicleEmissions(
-                            new Emissions(0d, 0d, 0d, 0d, 0d), sumoVehicle.lastVehicleData.getVehicleEmissions().getAllEmissions())
-                    );
+        if (isWaitingToLeaveParking) {
```

Comment:
Since `isWaitingToLeaveParking` only follows after a `isParking` or another `isWaitingToLeaveParking` state, the consumptions and emissions will always be 0 beforehand, so we can just copy the `VehicleData`